airbyte-cdk/python/airbyte_cdk/sources/declarative/yaml_declarative_source.py

```diff
@@ -17,9 +17,16 @@ def __init__(self, path_to_yaml):
 
     @property
     def connection_checker(self):
-        return self._factory.create_component(self._source_config["check"], dict())(source=self)
+        check = self._source_config["check"]
+        if "class_name" not in check:
+            check["class_name"] = "airbyte_cdk.sources.declarative.checks.check_stream.CheckStream"
```

Comment:
I don't think we want to prevent developers from overriding the check method, but we could potentially default to always checking either the first stream or all of them